### PR TITLE
Add test for "transferValueOnly" (async call and back transfer)

### PR DESCRIPTION
**Pull request by andreibancioiu**
This PR asserts the existence of an issue, which is fixed by: https://github.com/multiversx/mx-chain-go/pull/5936.

The issue refers to a duplicated `transferValueOnly` event, emitted by a contract call under certain conditions.

### CODE REVIEW

File: integrationTests/vm/wasm/transfers/transfers_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	require.Nil(t, err)
+
+	err = context.ExecuteSCWithValue(&context.Owner, "receive", big.NewInt(1))
+	require.Nil(t, err)
+	require.Len(t, context.LastLogs, 1)
+	require.Len(t, context.LastLogs[0].GetLogEvents(), 4)
+
+	events := context.LastLogs[0].GetLogEvents()
+
+	// There are duplicated events, to be fixed here:
```
Comment (iulianpascalau) [inline integrationTests/vm/wasm/transfers/transfers_test.go:27]: can we make this test fail as the current state of the `rc/v1.6.next1` and we will automatically merge this after #5936 ? The reference can remain

File: integrationTests/vm/wasm/utils.go [hunk 2]
```diff
@@ -550,12 +556,15 @@ func (context *TestContext) DeploySC(wasmPath string, parametersString string) e
...
 	}
 
-	err = context.GetCompositeTestError()
+	err = context.acquireOutcome()
 	if err != nil {
 		return err
 	}
 
-	_ = context.UpdateLastSCResults()
+	err = context.GetCompositeTestError()
```
Comment (iulianpascalau) [inline integrationTests/vm/wasm/utils.go:564]: return directly?

File: integrationTests/vm/wasm/utils.go [hunk 3]
```diff
@@ -604,12 +613,15 @@ func (context *TestContext) UpgradeSC(wasmPath string, parametersString string)
...
```
Comment (iulianpascalau) [inline integrationTests/vm/wasm/utils.go:621]: return directly?

File: integrationTests/vm/wasm/utils.go [hunk 4]
```diff
@@ -680,18 +692,21 @@ func (context *TestContext) ExecuteSCWithValue(sender *testParticipant, txData s
...
```
Comment (iulianpascalau) [inline integrationTests/vm/wasm/utils.go:700]: return directly?

File: integrationTests/vm/wasm/utils.go [hunk 5]
```diff
@@ -680,18 +692,21 @@ func (context *TestContext) ExecuteSCWithValue(sender *testParticipant, txData s
...
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
-// UpdateLastSCResults --
-func (context *TestContext) UpdateLastSCResults() error {
+// acquireOutcome --
```
Comment (iulianpascalau) [inline integrationTests/vm/wasm/utils.go:708]: remove one `-` ?

**Review by sstanculeanu — COMMENTED** [review]
no other comments

File: integrationTests/vm/wasm/utils.go (hunk 5 above)
Comment (andreibancioiu) [reply integrationTests/vm/wasm/utils.go]: Fixed.

File: integrationTests/vm/wasm/utils.go (hunk 4 above)
Comment (andreibancioiu) [reply integrationTests/vm/wasm/utils.go]: Fixed.

File: integrationTests/vm/wasm/utils.go (hunk 2 above)
Comment (andreibancioiu) [reply integrationTests/vm/wasm/utils.go]: Fixed.

File: integrationTests/vm/wasm/transfers/transfers_test.go (hunk 1 above)
Comment (andreibancioiu) [reply integrationTests/vm/wasm/transfers/transfers_test.go]: Indeed. Changed.

**Review by iulianpascalau — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**